scripts/cast-on-tenderly/index.js

```diff
@@ -19,63 +20,123 @@ if (!SPELL_ADDRESS) {
+            explorerPage: 'DISABLED',
+            syncState: true,
+        },
+    });
+    const testnetId = response.data.container.id;
+    const rpcEndpointPrivate = response.data.container.connectivityConfig.endpoints.find(
+        endpoint => endpoint.displayName === 'unlocked'
+    );
+    console.info(`tenderly testnet "${testnetId}" is created`);
+    return {
+        testnetId,
+        rpcUrlPrivate: rpcEndpointPrivate.uri,
+    };
 };
 
-const runSpell = async function () {
-    const { forkId, forkUrl, rpcUrl } = await createTenderlyFork();
-    console.info('private tenderly fork is created', forkUrl);
-
-    const provider = new ethers.providers.JsonRpcProvider(rpcUrl);
-    const signer = provider.getSigner();
+const publishTenderlyTestnet = async function (testnetId) {
+    console.info(`making tenderly testnet "${testnetId}" public...`);
+    const response = await makeTenderlyApiRequest({
+        method: 'put',
+        path: `/testnet/container/${testnetId}`,
+        body: {
+            explorerPage: 'ENABLED',
+        },
+    });
+    if (response.data?.container?.explorer_page !== 'ENABLED') {
+        throw new Error('failed to publish testnet');
+    }
+    console.info(`tenderly testnet is now public and discoverable`);
+    const rpcEndpointPublic = response.data.container.connectivityConfig.endpoints.find(
+        endpoint => endpoint.displayName === 'testnet'
+    );
+    const explorerUrlPublic = `https://dashboard.tenderly.co/explorer/vnet/${rpcEndpointPublic.id}`;
+    console.info(`public explorer url: ${explorerUrlPublic}`);
+    return { rpcUrlPrivate: rpcEndpointPublic.uri, explorerUrlPublic };
+};
 
+const makeSpellTheHat = async function (spellAddress, provider) {
     console.info('fetching the chief address from chainlog...');
     const chainlog = new Contract(
         CHAINLOG_ADDRESS,
         ['function getAddress(bytes32) external view returns (address)'],
-        signer
+        provider.getSigner()
     );
     const chiefAddress = await chainlog.getAddress(ethers.utils.formatBytes32String('MCD_ADM'));
 
     console.info('overwriting the hat...');
     await provider.send('tenderly_setStorageAt', [
         chiefAddress,
         ethers.utils.hexZeroPad(ethers.utils.hexValue(CHIEF_HAT_SLOT), 32),
-        ethers.utils.hexZeroPad(SPELL_ADDRESS, 32),
+        ethers.utils.hexZeroPad(spellAddress, 32),
     ]);
 
     console.info('checking the hat...');
-    const chief = new Contract(chiefAddress, ['function hat() external view returns (address)'], signer);
+    const chief = new Contract(chiefAddress, ['function hat() external view returns (address)'], provider.getSigner());
     const hatAddress = await chief.hat();
-    if (hatAddress !== SPELL_ADDRESS) {
+    if (hatAddress !== spellAddress) {
         throw new Error('spell does not have the hat');
     }
+    console.info('spell have the hat...');
+};
 
+const sheduleWarpAndCastSpell = async function (spellAddress, provider) {
```

Comment:
Nit: naming.
```suggestion
const castSpell = async function (spellAddress, provider) {
```

Reasoning: we are only interested in casting the spell. The fact that we need to schedule it first and warp time to be able to do it is an implementation detail.
Usually we want to try to name functions in terms of their intent, not the implementation details. For instance, you didn't name `publishTenderlyTestnet` `sendHttpRequestToTenderlyApiToPublishTestnet`.

Reply:
I disagree, the example is irrelevant. We're not doing anything inside `publishTenderlyTestnet` beyond publishing it. While in `castSpell` we're not only casing, but doing other high-level operations. If we can call it `executeSpell`, then it wouldn't at least imply a single operation of the same name.

But I would say unless we plan to reuse those functions outside of this file, the naming of the functions have very little practical sense.